assets/src/domain/eventEditor/ui/tickets/hooks/useDeleteTicketHandler.ts

```diff
@@ -1,36 +1,58 @@
 import { useCallback } from 'react';
 
 import useTicketPrices from '@edtrServices/apollo/queries/tickets/useTicketPrices';
-import { EntityId } from '@dataServices/types';
-import { useTicketMutator, usePriceMutator } from '@edtrServices/apollo/mutations';
+import type { EntityId } from '@dataServices/types';
+import { useTicketMutator } from '@edtrServices/apollo/mutations';
 import { getGuids } from '@appServices/predicates';
+import { usePriceQueryOptions } from '@edtrServices/apollo/queries';
+import type { PricesList } from '@edtrServices/apollo';
+import { useApolloClient } from '@apollo/react-hooks';
+import { assocPath } from 'ramda';
 
 type Callback = (deletePermanently?: boolean) => Promise<void>;
 
 const useDeleteTicketHandler = (id: EntityId): Callback => {
 	const { deleteEntity: deleteTicket } = useTicketMutator();
 	const relatedPrices = useTicketPrices(id);
-	const { deleteEntity: deletePrice } = usePriceMutator();
+	const priceQueryOptions = usePriceQueryOptions();
+	const client = useApolloClient();
 
-	const onDeleteTicket = useCallback<VoidFunction>(() => {
+	/**
+	 * Deletes the related prices from Apollo cache,
+	 * permanent deletion is handled server-side
+	 */
+	const deleteRelatedPrices = useCallback<VoidFunction>(() => {
 		// The prices that are not default or tax prices.
 		const pricesToDelete = relatedPrices.filter(({ isDefault, isTax }) => !isDefault && !isTax);
+		// if we have nothing to delete
+		if (!pricesToDelete.length) {
+			return;
+		}
 		const priceIdsToDelete = getGuids(pricesToDelete);
-		priceIdsToDelete.forEach((id) => {
-			deletePrice({ id, deletePermanently: true });
+		// read existing data from Apollo cache
+		const data = client.readQuery<PricesList>(priceQueryOptions);
+		// filter out the related prices from Apollo cache data
+		const pricesToRetain = data?.espressoPrices?.nodes?.filter(({ id }) => !priceIdsToDelete.includes(id)) || [];
```

Comment:
not sure about that `id` destructuring, in case when node is undefined;
what do you think of:
```
		const pricesToRetain = data?.espressoPrices?.nodes?.filter(node => !priceIdsToDelete.includes(node?id)) || [];
```

Reply:
If `nodes` is `undefined`, the filter won't be called :)

Reply:
![image](https://user-images.githubusercontent.com/18226415/83287918-230d5f00-a200-11ea-921f-bdb73bbfef03.png)

Reply:
just checked, as you wrote is safe enuf:

![image](https://user-images.githubusercontent.com/1477453/83287970-4aa5fc80-a1eb-11ea-841c-da8ddcbedfbd.png)

Reply:
mmmm... dunno if that's any clearer... 
considering it's passing the `id` to `priceIdsToDelete` i think it can easily be inferred that those are price IDs
imho

Reply:
Yes, optional chaining "breaks the chain" whenever it finds `null` or `undefined`

Reply:
I think more extracting might be clearer:
```
const prices = data?.espressoPrices?.nodes;
const pricesToRetain = prices?.filter(({ id }) => !priceIdsToDelete.includes(id)) ?? [];
```